pypdf/generic/_data_structures.py

```diff
@@ -1256,7 +1256,7 @@ def isolate_graphics_state(self) -> None:
             self._operations.insert(0, ([], "q"))
             self._operations.append(([], "Q"))
         elif self._data:
-            self._data = b"q\n" + b_(self._data) + b"Q\n"
+            self._data = b"q\n" + b_(self._data) + b"\nQ"
```

Comment:
```suggestion
            self._data = b"q\n" + b_(self._data) + b"\nQ\n"
```

We should still keep the final newline here.

Reply:
The output looked a bit nicer in the samples I was looking at without also adding the trailing newline, but it works for me either way - I'm not very familiar with pdf internals